### PR TITLE
Import Role Grants

### PR DESCRIPTION
**Pull request by alldoami**

### CODE REVIEW

File: providers/snowflake/role_grant.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	type tfGrant struct {
+		Role      string
+		Privilege string
+		Roles     []string
+		Shares    []string
+	}
+	groupedResources := map[string]*tfGrant{}
+	for _, grant := range roleGrantList {
+		// TODO(ad): Fix this csv delimited when fixed in the provider. We should use the same functionality.
+		id := grant.Name.String
```
Comment (edulop91) [inline providers/snowflake/role_grant.go]: these probably have a different ID

File: providers/snowflake/snowflake_client.go [hunk 2]
```diff
@@ -274,3 +285,21 @@ func (sc *client) ListSchemaGrants(database database, schema schema) ([]schemaGr
 	}
 	return schemaGrants, errors.Wrap(err, "unable to scan row for SHOW GRANTS ON SCHEMA")
 }
+
+func (sc *client) ListRoleGrants(role role) ([]roleGrant, error) {
+	sdb := sqlx.NewDb(sc.db, "snowflake")
+	stmt := fmt.Sprintf(`SHOW GRANTS ON ROLE "%v"`, role.Name.String)
+	fmt.Println(stmt)
```
Comment (edulop91) [inline providers/snowflake/snowflake_client.go]: rm or make a debug statement

File: providers/snowflake/role_grant.go (hunk 1 above)
Comment (alldoami) [reply providers/snowflake/role_grant.go]: https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/master/pkg/resources/role_grants.go#L73

Comment (edulop91) [reply providers/snowflake/role_grant.go]: oh, you're right!

File: providers/snowflake/role_grant.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+func (g RoleGrantGenerator) createResources(roleGrantList []roleGrant) []terraform_utils.Resource {
+	type tfGrant struct {
+		Role      string
+		Privilege string
+		Roles     []string
+		Shares    []string
+	}
+	groupedResources := map[string]*tfGrant{}
+	for _, grant := range roleGrantList {
+		// TODO(ad): Fix this csv delimited when fixed in the provider. We should use the same functionality.
```
Comment (edulop91) [inline providers/snowflake/role_grant.go]:
```suggestion
```
that comment not needed then

File: providers/snowflake/role_grant.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		}
+		if grant.GrantedTo.String == "SHARE" {
+			tfGrant.Shares = append(tfGrant.Shares, grant.GranteeName.String)
+		}
+	}
+	var resources []terraform_utils.Resource
+	for id, grant := range groupedResources {
+		resources = append(resources, terraform_utils.NewResource(
+			id,
+			fmt.Sprintf("%v_%v", grant.Role, grant.Privilege),
```
Comment (edulop91) [inline providers/snowflake/role_grant.go]:
```suggestion
			fmt.Sprintf("%s_%s", grant.Role.String, grant.Privilege.String),
```

File: providers/snowflake/schema_grant.go [hunk 5]
```diff
@@ -33,7 +34,9 @@ func (g SchemaGrantGenerator) createResources(schemaGrantList []schemaGrant) []t
 	groupedResources := map[string]*tfGrant{}
 	for _, grant := range schemaGrantList {
 		// TODO(ad): Fix this csv delimited when fixed in the provider. We should use the same functionality.
-		id := fmt.Sprintf("%v|||%v", grant.Name.String, grant.Privilege.String)
+		DB := strings.Split(grant.Name.String, ".")[0]
+		Sc := strings.Split(grant.Name.String, ".")[1]
```
Comment (edulop91) [inline providers/snowflake/schema_grant.go]: sc not very descriptive/kind of confusing -- might need a better name

File: providers/snowflake/schema.go [hunk 6]
```diff
@@ -24,9 +26,14 @@ type SchemaGenerator struct {
 func (g SchemaGenerator) createResources(schemaList []schema) []terraform_utils.Resource {
 	var resources []terraform_utils.Resource
 	for _, schema := range schemaList {
+		name := schema.Name.String
+		if name == "INFORMATION_SCHEMA" || name == "PUBLIC" {
```
Comment (edulop91) [inline providers/snowflake/schema.go]: maybe leave a comment why
"These schemas are always managed by snowflake, skp them" or something like that

File: providers/snowflake/schema.go [hunk 7]
```diff
@@ -24,9 +26,14 @@ type SchemaGenerator struct {
...
+			continue
+		}
 		resources = append(resources, terraform_utils.NewSimpleResource(
-			schema.Name.String,
-			schema.Name.String,
+			// Schemas need to be namespaced by their database with two underscores
+			fmt.Sprintf("%v__%v", schema.DatabaseName.String, schema.Name.String),
```
Comment (edulop91) [inline providers/snowflake/schema.go]:
```suggestion
			fmt.Sprintf("%s__%s", schema.DatabaseName.String, schema.Name.String),
```
%s for strings

File: providers/snowflake/schema.go [hunk 8]
```diff
@@ -24,9 +26,14 @@ type SchemaGenerator struct {
...
+			fmt.Sprintf("%v__%v", schema.DatabaseName.String, schema.Name.String),
```
Comment (edulop91) [inline providers/snowflake/schema.go]:
```suggestion
			fmt.Sprintf("%s__%s", schema.DatabaseName.String, schema.Name.String),
```

**Pull request merged**